xtc/src/ledger.rs

```diff
@@ -481,3 +596,12 @@ mod tests {
         assert_eq!(ledger.balance(&charlie()), 0);
     }
 }
+
+#[update]
+pub async fn balance(account: Option<Principal>) -> u64 {
+    let ic = get_context();
+    let caller = ic.caller();
+    crate::progress().await;
+    let ledger = ic.get::<Ledger>();
+    ledger.balance(&account.unwrap_or(caller))
```

Comment:
warning: use of deprecated function `ic_kit::get_context`: get_context is deprecated use ic_kit::ic::*

```suggestion
    let caller = ic::caller();
    crate::progress().await;
    let ledger = ic::get::<Ledger>();
    ledger.balance(&account.unwrap_or(caller))
```